File: katas/7 kyu/summing-a-numbers-digits/solution.py
# summing-a-numbers-digits
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def sum_digits(number):
  return sum([int(d) for d in str(number) if d.isnumeric()])
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(sum_digits(10), 1)
test.assert_equals(sum_digits(99), 18)
test.assert_equals(sum_digits(-32), 5)

File: katas/7 kyu/summing-a-numbers-digits/solution.js
// summing-a-numbers-digits
/* ------------------------------- MY SOLUTION ------------------------------ */
function sumDigits(number) {
  return ("" + Math.abs(number)).split("").reduce((a , b) => Number(a) + Number(b) , 0)
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const { assert } = require("chai")

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(sumDigits(10), 1);
    assert.strictEqual(sumDigits(99), 18);
    assert.strictEqual(sumDigits(-32), 5);
    })
  })